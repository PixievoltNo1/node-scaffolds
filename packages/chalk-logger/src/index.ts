import type { Mutable } from '@guanghechen/utility-types'
import { partOptionsFromArgs } from './commander'
import type { Level } from './level'
import type { ILoggerOptions } from './logger'
import { Logger } from './logger'

export * from './color'
export * from './commander'
export * from './level'
export * from './logger'

export class ChalkLogger extends Logger {
  /**
   * prefix of logger.name
   */
  protected basename: string | null = null
  /**
   * name passed into .setName()
   */
  protected divisionName: string | null = null

  constructor(options?: ILoggerOptions, args?: string[]) {
    const optionsFromArgs = partOptionsFromArgs(args || [])
    const resolvedOptions = {
      ...options,
      ...optionsFromArgs,
      flags: { ...options?.flags, ...optionsFromArgs.flags },
    }

    super(resolvedOptions)
    const basename = resolvedOptions?.name ?? ''
    this.setBaseName(basename)
  }

  /**
   * update logger's level
   * @param level
   */
  public setLevel(level: Level | null | undefined): void {
    if (level == null) return
    const self = this as Mutable<this>
    self.level = level
  }

  /**
   * update logger's name
   * @param name
   */
  public setName(name: string | null): void {
    const resolvedName: string = [this.basename, name]
      .filter((x): x is string => x != null && x.length > 0)
      .join(' ')
    const self = this as Mutable<this>
    self.name = resolvedName
    this.divisionName = name
  }

  /**
   * update basename of logger
   * @param basename
   */
  public setBaseName(basename: string | null): void {
    this.basename = basename
    this.setName(this.divisionName)
  }

  /**
   * update logger's mode
   * @param mode
   */
  public setMode(mode: 'normal' | 'loose'): void {
    const self = this as Mutable<this>
    self.mode = mode
  }
}

export default ChalkLogger
