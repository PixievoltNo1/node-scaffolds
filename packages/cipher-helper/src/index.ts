export * from './_base'
export * from './aes'
export * from './catalog'
export * from './types/catalog'
export * from './types/cipher-helper'
export * from './util/buffer'
export * from './util/key'
