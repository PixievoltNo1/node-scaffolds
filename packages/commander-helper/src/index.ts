export * from './command'
export * from './fs'
export * from './git'
export * from './merge'
export * from './option'
export * from './yarn'
