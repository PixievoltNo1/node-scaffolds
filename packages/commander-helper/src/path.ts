import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetPath?: string | null,
): string {
  if (targetPath == null) return workspace
  const absoluteDir: string = path.resolve(workspace, targetPath)
  return absoluteDir
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}
