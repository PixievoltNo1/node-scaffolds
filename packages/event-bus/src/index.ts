export * from './SimpleEventBus'
export * from './types'
