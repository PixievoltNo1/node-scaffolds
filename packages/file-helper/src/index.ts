export * from './big-file/helper'
export * from './big-file/util'
export * from './big-file/types'
export * from './util'
