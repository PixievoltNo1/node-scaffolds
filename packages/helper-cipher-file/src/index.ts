export * from './types/IFileCipher'
export * from './types/IFileCipherBatcher'
export * from './types/IFileCipherFactory'
export * from './util/fingerprint'

export * from './FileCipher'
export * from './FileCipherBatcher'
export * from './FileCipherCatalog'
export * from './FileCipherCatalog.readonly'
export * from './FileCipherFactory'
