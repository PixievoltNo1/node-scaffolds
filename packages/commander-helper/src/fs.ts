import { isFileSync } from '@guanghechen/file-helper'
import fs from 'fs-extra'
import yaml from 'js-yaml'
import path from 'path'

/**
 * Load configuration file with format .json / .yml / .yaml
 *
 * @param filepath
 * @param logger
 */
export async function loadJsonOrYaml(
  filepath: string,
  encoding = 'utf-8',
): Promise<unknown | never> {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): Promise<string> =>
    fs.readFile(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = await loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = await loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}

/**
 * Load configuration file with format .json / .yml / .yaml  (synchronizing)
 *
 * @param filepath
 * @param logger
 */
export function loadJsonOrYamlSync(
  filepath: string,
  encoding = 'utf-8',
): unknown | never {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): string =>
    fs.readFileSync(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}
