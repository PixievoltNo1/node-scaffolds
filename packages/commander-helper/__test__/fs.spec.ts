import { desensitize, locateFixtures } from 'jest.setup'
import { collectAllFilesSync, loadJsonOrYaml, loadJsonOrYamlSync } from '../src'

describe('loadJsonOrYaml', function () {
  test('.json', async function () {
    expect(
      await loadJsonOrYaml(locateFixtures('basic/config.json')),
    ).toMatchSnapshot()
  })

  test('.yml', async function () {
    expect(
      await loadJsonOrYaml(locateFixtures('basic/config.yml')),
    ).toMatchSnapshot()
  })

  test('.yaml', async function () {
    expect(
      await loadJsonOrYaml(locateFixtures('basic/config.yaml')),
    ).toMatchSnapshot()
  })

  test('unknown filetype', async function () {
    await expect(() =>
      loadJsonOrYaml(locateFixtures('basic/config.json-unknown')),
    ).rejects.toThrow(
      'Only files in .json / .yml / .ymal format are supported.',
    )
  })

  test('non existent filepath', async function () {
    await expect(() =>
      loadJsonOrYaml(locateFixtures('basic/config.json-non-exist')),
    ).rejects.toThrow('is an invalid file path')
  })
})

describe('loadJsonOrYamlSync', function () {
  test('.json', function () {
    expect(
      loadJsonOrYamlSync(locateFixtures('basic/config.json')),
    ).toMatchSnapshot()
  })

  test('.yml', function () {
    expect(
      loadJsonOrYamlSync(locateFixtures('basic/config.yml')),
    ).toMatchSnapshot()
  })

  test('.yaml', function () {
    expect(
      loadJsonOrYamlSync(locateFixtures('basic/config.yaml')),
    ).toMatchSnapshot()
  })

  test('unknown filetype', function () {
    expect(() =>
      loadJsonOrYamlSync(locateFixtures('basic/config.json-unknown')),
    ).toThrow('Only files in .json / .yml / .ymal format are supported.')
  })

  test('non existent filepath', function () {
    expect(() =>
      loadJsonOrYamlSync(locateFixtures('basic/config.json-non-exist')),
    ).toThrow('is an invalid file path')
  })
})

describe('collectAllFilesSync', function () {
  test('null predicate', function () {
    expect(
      desensitize(collectAllFilesSync(locateFixtures('basic'), null)),
    ).toMatchSnapshot()
  })

  test('json file only', function () {
    expect(
      desensitize(
        collectAllFilesSync(locateFixtures('basic'), p => /\.json$/.test(p)),
      ),
    ).toMatchSnapshot()
  })

  test('collect from file', function () {
    expect(
      desensitize(
        collectAllFilesSync(locateFixtures('basic/config.json'), null),
      ),
    ).toMatchSnapshot()
  })
})
