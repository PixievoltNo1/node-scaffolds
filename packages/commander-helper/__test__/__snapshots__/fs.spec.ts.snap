// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectAllFilesSync collect from file 1`] = `
Array [
  "<$WORKSPACE$>/packages/commander-helper/__test__/fixtures/basic/config.json",
]
`;

exports[`collectAllFilesSync json file only 1`] = `
Array [
  "<$WORKSPACE$>/packages/commander-helper/__test__/fixtures/basic/config.json",
]
`;

exports[`collectAllFilesSync null predicate 1`] = `
Array [
  "<$WORKSPACE$>/packages/commander-helper/__test__/fixtures/basic/child/a.txt",
  "<$WORKSPACE$>/packages/commander-helper/__test__/fixtures/basic/config.json",
  "<$WORKSPACE$>/packages/commander-helper/__test__/fixtures/basic/config.json-unknown",
  "<$WORKSPACE$>/packages/commander-helper/__test__/fixtures/basic/config.yaml",
  "<$WORKSPACE$>/packages/commander-helper/__test__/fixtures/basic/config.yml",
]
`;

exports[`loadJsonOrYaml .json 1`] = `
Object {
  "name": "ghc-cli",
  "version": "1.0.11",
}
`;

exports[`loadJsonOrYaml .yaml 1`] = `
Object {
  "name": "ghc-cli",
  "version": "1.0.1",
}
`;

exports[`loadJsonOrYaml .yml 1`] = `
Object {
  "name": "ghc-cli",
  "version": "1.0.1",
}
`;

exports[`loadJsonOrYamlSync .json 1`] = `
Object {
  "name": "ghc-cli",
  "version": "1.0.11",
}
`;

exports[`loadJsonOrYamlSync .yaml 1`] = `
Object {
  "name": "ghc-cli",
  "version": "1.0.1",
}
`;

exports[`loadJsonOrYamlSync .yml 1`] = `
Object {
  "name": "ghc-cli",
  "version": "1.0.1",
}
`;
